reflect_test.go

```diff
@@ -100,18 +103,69 @@ func TestSchemaGeneration(t *testing.T) {
 				return
 			}
 
+			expectedSchema.Type = reconcileTypes(expectedSchema.Type)
+			expectedSchema.Definitions = reconcileEnumTypes(expectedSchema.Definitions)
+
 			if !reflect.DeepEqual(actualSchema, expectedSchema) {
-				actualJSON, err := json.MarshalIndent(actualSchema, "", "  ")
+				actualJSON, err := json.Marshal(actualSchema)
 				if err != nil {
 					t.Errorf("json.MarshalIndent(%v, \"\", \"  \"): %v", actualSchema, err)
 					return
 				}
-				t.Errorf("reflector %+v wanted schema %s, got %s", tt.reflector, f, actualJSON)
+
+				expectedJSON, err := json.Marshal(expectedSchema)
+				if err != nil {
+					t.Errorf("json.MarshalIndent(%v, \"\", \"  \"): %v", expectedJSON, err)
+					return
+				}
+
+				t.Errorf("reflector %+v wanted schema %s, got %s", tt.reflector, expectedJSON, actualJSON)
 			}
 		})
 	}
 }
 
+// The marshaling of json into interface{} results in a mismatch when we DeepEqual the expected/actual schemas for enum
+// This coerces all float64 enums back to int
+func reconcileTypes(t *Type) *Type {
+	t.Definitions = reconcileEnumTypes(t.Definitions)
+	t.Properties = convertEnum(t.Properties)
+
+	return t
+}
+
+func reconcileEnumTypes(definitions Definitions) Definitions {
+	mismatched := convertDefinitions(definitions)
+	converted :=  convertEnum(mismatched)
+
+	return Definitions(converted)
+}
+
+func convertDefinitions(definitions Definitions) map[string]*Type {
+	return map[string]*Type(definitions)
+}
+
+func convertEnum(definitions map[string]*Type) map[string]*Type {
+	for _, v := range definitions {
+		if len(v.Definitions) > 0 {
+			d := convertDefinitions(v.Definitions)
+			v.Definitions = convertEnum(d)
+		}
+
+		if len(v.Properties) > 0 {
+			v.Properties = convertEnum(v.Properties)
+		}
+
+		if len(v.Enum) > 0 && (reflect.TypeOf(v.Enum[0]).Kind() == reflect.Float64) {
+			for idx, val := range v.Enum {
+				v.Enum[idx] = int(val.(float64))
```

Comment:
shouldn't we do this the other way around? If unmarshalling is going to convert into a float64, then we should convert these all into float64, not ints

Reply:
No because there is no reason for json to be unmarshaled into `Schema` other than for the purpose of testing. Honestly I was just following the existing convention established in `numbericKeywords` since it's using `Atoi`, but that means these tags are actually busted for floats. The real fix will need to split or have `numberKeywords` handle both types, but will still need the code here that needs to be adjusted to apply only for `integer`.